app/rust/src/parser/c32.rs

```diff
@@ -64,55 +66,75 @@ fn double_sha256_checksum(data: &mut [u8; SHA256_LEN]) {
 fn double_sha256_checksum(data: &mut [u8; SHA256_LEN]) {
     let mut output = [0u8; SHA256_LEN];
     // safe to unwrap as we are passing the right len
-    sha256(&data[..21], &mut output[..]).unwrap();
-    data.copy_from_slice(output.as_ref());
+    sha256(&data[..21], &mut output[..]).apdu_unwrap();
+    data.iter_mut()
```

Comment:
Is this really better ? I doubt there's a panic in there considering both slice references come from arrays, BUT if it's still there how about `(data.len() != output.len()).then_some(()).apdu_unwrap();`? (there are nicer/longer alternatives, like an explicit if with unreachable_unchecked inside)

Reply:
the idea was to remove calls to copy_from_slice for which the compiler would add the panic machinery to handle that case, but after looking at the result from compiler explorer, the copy_from_slice is simpler in terms of amount of instructions.